modular/executor/execute_replicate.go

```diff
@@ -122,13 +125,26 @@ func (e *ExecuteModular) handleReplicatePiece(ctx context.Context, rTask coretas
 	}
 	doneReplicate := func(ctx context.Context) error {
 		log.Debug("start to done replicate")
-		for rIdx, sp := range rTask.GetSecondaryEndpoints() {
-			log.Debugw("start to done replicate", "sp", sp)
-			signature, innerErr := e.doneReplicatePiece(ctx, rTask, sp, int32(rIdx))
+		var gvg *virtualgrouptypes.GlobalVirtualGroup
+		gvg, err = e.baseApp.GfSpClient().GetGlobalVirtualGroupByGvgID(ctx, rTask.GetGlobalVirtualGroupId())
+		if err != nil {
+			return ErrConsensusWithDetail("QueryGVGInfo error: " + err.Error())
+		}
+		for rIdx, spEp := range rTask.GetSecondaryEndpoints() {
+			log.Debugw("start to done replicate", "sp", spEp)
+			signature, innerErr := e.doneReplicatePiece(ctx, rTask, spEp, int32(rIdx))
 			if innerErr == nil {
+				msg := storagetypes.NewSecondarySpSealObjectSignDoc(e.baseApp.ChainID(), gvg.Id, rTask.GetObjectInfo().Id, storagetypes.GenerateHash(rTask.GetObjectInfo().GetChecksums()[:])).GetBlsSignHash()
+				err = veritySecondarySpBlsSignature(e.getSpByID(gvg.GetSecondarySpIds()[rIdx]), signature, msg[:])
+				if err != nil {
+					rTask.SetNotAvailableSpIdx(int32(rIdx))
+					log.CtxErrorw(ctx, "verify secondary SP bls signature failed", "secondary_sp_id", gvg.GetSecondarySpIds()[rIdx], "error", err.Error())
```

Comment:
fix as `failed to verify xxx` is better.

Reply:
fixed